### PR TITLE
Add helm-step-delete, helm-step-list

### PR DESCRIPTION
**Pull request by rick-a-lane-ii**
- Adds `helm-step-delete`
- Adds `helm-step-list`
- Minor cleanup

### CODE REVIEW

File: steps/delete/spec.schema.json [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      },
+      "required": [
+        "ca.pem",
+        "key.pem",
+        "cert.pem"
+      ]
+    },
+    "purge": {
+      "type": "boolean",
+      "description": "If `true`, all release information will be purged. (default: false)",
```
Comment (impl) [inline steps/delete/spec.schema.json]: FWIW you don't have to put the default in the description in this case. I think our UI can render the defaults appropriately.

File: steps/delete/spec.schema.json [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      "default": false
+    }
+  },
+  "required": [
+    "credentials",
```
Comment (impl) [inline steps/delete/spec.schema.json]: I believe Helm credentials are actually not required for some (insecure-on-purpose) Tiller installations.

**Review by impl — APPROVED** [review]
LGTM

File: steps/delete/spec.schema.json (hunk 1 above)
Comment (rick-a-lane-ii) [reply steps/delete/spec.schema.json]: Yeah, that's a copy/paste issue. I've seen that done in a few places, and I didn't really want to make a judgment call on whether that's necessary or not. I'll clean up this repo though.

File: steps/delete/spec.schema.json (hunk 2 above)
Comment (rick-a-lane-ii) [reply steps/delete/spec.schema.json]: Yeah, I know, I wrote all of the original logic to handle both scenarios.

**Pull request merged**